BootloaderCorePkg/Stage2/Stage2Support.c

```diff
@@ -403,33 +404,26 @@ BuildBaseInfoHob (
     SerialPortInfo->UartPciAddr = 0;
   }
 
-  // Build Frame Buffer hob
-  FspGfxHob = (EFI_PEI_GRAPHICS_INFO_HOB *)GetGuidHobData (LdrGlobal->FspHobList, &Length,
+  // Build graphic info hob
+  BlGfxHob = BuildGuidHob (&gEfiGraphicsInfoHobGuid, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
+  ZeroMem (BlGfxHob, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
+  if (BlGfxHob != NULL) {
+    FspGfxHob = (EFI_PEI_GRAPHICS_INFO_HOB *)GetGuidHobData (LdrGlobal->FspHobList, &Length,
               &gEfiGraphicsInfoHobGuid);
-  if (FspGfxHob != NULL) {
-    FrameBuffer.LinearFrameBuffer    = FspGfxHob->FrameBufferBase;
-    FrameBuffer.HorizontalResolution = FspGfxHob->GraphicsMode.HorizontalResolution;
-    FrameBuffer.VerticalResolution   = FspGfxHob->GraphicsMode.VerticalResolution;
-    FrameBuffer.BitsPerPixel         = 32;
-    FrameBuffer.BytesPerScanLine     = (UINT16) (FspGfxHob->GraphicsMode.PixelsPerScanLine * 4);
-    FrameBuffer.Red.Position         = 0x10;
-    FrameBuffer.Red.Mask             = 0xFF;
-    FrameBuffer.Green.Position       = 0x08;
-    FrameBuffer.Green.Mask           = 0xFF;
-    FrameBuffer.Blue.Position        = 0x00;
-    FrameBuffer.Blue.Mask            = 0xFF;
-    FrameBuffer.Reserved.Position    = 0x18;
-    FrameBuffer.Reserved.Mask        = 0xFF;
-  } else {
-    FrameBuffer.LinearFrameBuffer    = 0;
+    if (FspGfxHob != NULL) {
+      CopyMem (BlGfxHob, FspGfxHob, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
+      DEBUG ((DEBUG_INFO, "Graphics Info: %d x %d x 32 @ 0x%08X\n", \
+        BlGfxHob->GraphicsMode.HorizontalResolution,\
+        BlGfxHob->GraphicsMode.VerticalResolution,\
+        BlGfxHob->FrameBufferBase));
+    } else {
+      DEBUG ((DEBUG_INFO, "Failed to get Graphics Info HOB from FSP\n"));
+    }
   }
 
-  if (FrameBuffer.LinearFrameBuffer) {
-    FrameBufferInfo = BuildGuidHob (&gLoaderFrameBufferInfoGuid, sizeof (FRAME_BUFFER_INFO));
-    CopyMem (FrameBufferInfo, &FrameBuffer, sizeof (FRAME_BUFFER_INFO));
-    DEBUG ((DEBUG_INFO, "Framebuffer Info: %d x %d x 32 @ 0x%08X\n", \
-            FrameBuffer.HorizontalResolution, FrameBuffer.VerticalResolution, FrameBuffer.LinearFrameBuffer));
-  }
+  // Build graphic device info hob
+  BlGfxDeviceInfo = BuildGuidHob (&gEfiGraphicsDeviceInfoHobGuid, sizeof (EFI_PEI_GRAPHICS_DEVICE_INFO_HOB));
```

Comment:
Should we check NULL for BlGfxDeviceInfo?

Reply:
will add NULL check.